iis-wf/iis-wf-export-actionmanager/src/test/java/eu/dnetlib/iis/wf/export/actionmanager/sequencefile/FieldAccessorTest.java

```diff
@@ -0,0 +1,145 @@
+package eu.dnetlib.iis.wf.export.actionmanager.sequencefile;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.cxf.helpers.IOUtils;
+import org.junit.Test;
+
+import com.googlecode.protobuf.format.JsonFormat;
+
+import eu.dnetlib.actionmanager.actions.AtomicAction;
+import eu.dnetlib.actionmanager.common.Agent;
+import eu.dnetlib.data.proto.OafProtos;
+
+/**
+ * Field accessor test class.
+ * @author mhorst
+ *
+ */
+public class FieldAccessorTest {
+
+	@Test
+	public void testAccessingValuesUsingDecoder() throws Exception {
+		FieldAccessor accessor = new FieldAccessor();
+		accessor.registerDecoder("TargetValue", new OafFieldDecoder());
+
+		String rawSetId = "rawset-id";
+		String agentId = "agent-id";
+		String agentName = "agent-name";
+		AtomicAction action = new AtomicAction(rawSetId, new Agent(agentId, agentName, Agent.AGENT_TYPE.service));
+		byte[] decodedTargetValue = decodeTargetValue(new String(IOUtils.readBytesFromStream(FieldAccessorTest.class
+				.getResourceAsStream("/eu/dnetlib/iis/wf/export/actionmanager/sequencefile/resultProject.json")),
+				"utf8"));
+		action.setTargetValue(decodedTargetValue);
+
+		assertEquals(rawSetId, accessor.getValue("RawSet", action).toString());
+		assertEquals(agentId, accessor.getValue("Agent.Id", action).toString());
+		assertEquals(agentName, accessor.getValue("Agent.Name", action).toString());
+		assertEquals(Agent.AGENT_TYPE.service.toString(), accessor.getValue("Agent.Type", action).toString());
+		assertEquals("relation", accessor.getValue("$TargetValue.Kind", action).toString());
+		assertEquals("resultProject", accessor.getValue("$TargetValue.Rel.RelType", action).toString());
+	}
+
+	@Test(expected=Exception.class)
+	public void testAccessingValuesForInvalidPath() throws Exception {
+		FieldAccessor accessor = new FieldAccessor();
+		accessor.registerDecoder("TargetValue", new OafFieldDecoder());
+
+		String rawSetId = "rawset-id";
+		String agentId = "agent-id";
+		String agentName = "agent-name";
+		AtomicAction action = new AtomicAction(rawSetId, new Agent(agentId, agentName, 
+				Agent.AGENT_TYPE.service));
+
+		accessor.getValue("$TargetValue.Unknown", action).toString();
```

Comment:
Yeap. This is because `$TargetValue` is enough for the test to pass as it is not present in action. What makes test fail is looking e.g. for `RawSet` or `Agent.Id` which is accessible.

I changed  `$TargetValue.Unknown` into `Agent.Unknown` to avoid confusions.